api_tests/src/actors/balance_asserter.ts

```diff
@@ -0,0 +1,194 @@
+import { EthereumWallet } from "../wallets/ethereum";
+import { BitcoinWallet } from "../wallets/bitcoin";
+import { LightningWallet } from "../wallets/lightning";
+
+export interface BalanceAsserter {
+    assertSwapped(): Promise<void>;
+    assertRefunded(): Promise<void>;
+    assertNothingReceived(): Promise<void>;
+}
+
+export class Erc20BalanceAsserter implements BalanceAsserter {
+    public static async forFunder(
+        wallet: EthereumWallet,
+        swapAmount: bigint,
+        tokenContract: string
+    ) {
+        await wallet.mintErc20(swapAmount, tokenContract); // As funder, we need money to spend
+
+        const startingBalance = await wallet.getErc20Balance(tokenContract, 0);
```

Comment:
What is the `0` that is passed as second parameter? Maybe we can be more explicit.

Reply:
That is the number of decimal points of the the contract. I think we can remove that now, good catch!